Assets/Scripts/Player/PlayerController.cs

```diff
@@ -83,30 +102,106 @@ private void Update()
+        /// Is called on mouse click.
+        /// If right clicked, checks whether clicked on a damageable object or the ground.
+        /// If clicked on a damageable object, attack it. If clicked on the ground, cancel attack
+        /// and move to the clicked point.
         /// </summary>
-        private void Move(InputAction.CallbackContext obj)
+        private void OnRightClick(InputAction.CallbackContext obj)
         {
             Ray clickRay = _camera.ScreenPointToRay(Mouse.current.position.ReadValue());
 
-            // Get ground position from mouse click
-            if (Physics.Raycast(clickRay, out RaycastHit hit, 10000, moveClickLayers))
+            // only change target / move, if not performing a hit
+            if (Physics.Raycast(clickRay, out RaycastHit hit, 10000) && !_attackLogic.PerformingHit)
             {
-                // Set destination for nav mesh agent
-                _navMeshAgent.SetDestination(hit.point);
-                _navMeshAgent.isStopped = false;
-
-                // Create click point effect
-                Instantiate(clickEffect, hit.point + Vector3.up * 5, Quaternion.identity);
+                GameObject objectHit = hit.collider.gameObject;
+                IDamageable damageable = (IDamageable)objectHit.GetComponent<IDamageable>();
+                if (damageable != null)
+                {
+                    // implementation NOT capable of area damage
+                    _attackLogic.StartAttack(objectHit);
+                }
+                // Get ground position from mouse click
+                else if (Physics.Raycast(clickRay, out RaycastHit groundHit, 10000, moveClickLayers))
+                {
+                    // cancel possible ongoing attacks
+                    _attackLogic.StopAttack();
+                    Move(groundHit);
+                }
             }
         }
 
+        /// <summary>
+        /// Player moves to the pressed location using the NavMeshAgent.
+        /// <param name="hit">The point on the ground the player should move to.</param>
+        /// </summary>
+        private void Move(RaycastHit hit)
+        {
+            // Set destination for nav mesh agent
+            _navMeshAgent.SetDestination(hit.point);
+            _navMeshAgent.isStopped = false;
+
+            // Create click point effect
+            Instantiate(clickEffect, hit.point + Vector3.up * 5, Quaternion.identity);
+        }
+
+        /// <summary>
+        /// Faces the target. Returns true if facing the target.
+        /// </summary>
+        /// <param name="target">The target to face</param>
+        /// <param name="shouldTurn">Whether the object should turn to the target or not</param>
+        /// <returns>Whether the object is facing the target</returns>
+        public bool FaceTarget(GameObject target, bool shouldTurn)
+        {
+            float f;
+            return FaceTarget(target, shouldTurn, out f);
```

Comment:
```suggestion
            return FaceTarget(target, shouldTurn, out _);
```
Then you can delete the local variable